backend-tests/tests/test_devauth.py

```diff
@@ -12,22 +12,1401 @@
+        returns DevWithAuthsets objects."""
+    useradmm = ApiClient(useradm.URL_MGMT)
+
+    # log in user
+    r = useradmm.call("POST", useradm.URL_LOGIN, auth=(user.name, user.pwd))
+    assert r.status_code == 200
+
+    utoken = r.text
+
+    devices = []
+
+    # some vanilla 'pending' devices, single authset
+    for _ in range(5):
+        dev = make_pending_device(utoken, 1, tenant_token=tenant_token)
+        devices.append(dev)
+
+    # some pending devices with > 1 authsets
+    for i in range(2):
+        dev = make_pending_device(utoken, 3, tenant_token=tenant_token)
+        devices.append(dev)
+
+    # some 'accepted' devices, single authset
+    for _ in range(3):
+        dev = make_accepted_device_with_multiple_authsets(
+            utoken, 1, tenant_token=tenant_token
+        )
+        devices.append(dev)
+
+    # some 'accepted' devices with >1 authsets
+    for _ in range(2):
+        dev = make_accepted_device_with_multiple_authsets(
+            utoken, 3, tenant_token=tenant_token
+        )
+        devices.append(dev)
+
+    # some rejected devices
+    for _ in range(2):
+        dev = make_rejected_device(utoken, 3, tenant_token=tenant_token)
+        devices.append(dev)
+
+    # preauth'd devices
+    for i in range(2):
+        dev = make_preauthd_device(utoken)
+        devices.append(dev)
+
+    # preauth'd devices with extra 'pending' sets
+    for i in range(2):
+        dev = make_preauthd_device_with_pending(
+            utoken, num_pending=2, tenant_token=tenant_token
+        )
+        devices.append(dev)
+    devices.sort(key=lambda dev: dev.id)
+    return devices
+
+
+@pytest.yield_fixture(scope="function")
+def devs_authsets(user):
+    yield make_devs_with_authsets(user)
+
+
+@pytest.yield_fixture(scope="function")
+def tenants_devs_authsets(tenants_users):
+    for t in tenants_users:
+        devs = make_devs_with_authsets(t.users[0], t.tenant_token)
+        t.devices = devs
+
+    yield tenants_users
+
+
+def rand_id_data():
+    mac = ":".join(["{:02x}".format(random.randint(0x00, 0xFF), "x") for i in range(6)])
+    sn = "".join(["{}".format(random.randint(0x00, 0xFF)) for i in range(6)])
+
+    return {"mac": mac, "sn": sn}
+
+
+def make_pending_device(utoken, num_auth_sets=1, tenant_token=""):
+    devauthm = ApiClient(deviceauth.URL_MGMT)
+    devauthd = ApiClient(deviceauth.URL_DEVICES)
+
+    id_data = rand_id_data()
+
+    dev = None
+    for i in range(num_auth_sets):
+        priv, pub = testutils.util.crypto.rsa_get_keypair()
+        new_set = create_authset(
+            devauthd, devauthm, id_data, pub, priv, utoken, tenant_token=tenant_token
+        )
+
+        if dev is None:
+            dev = Device(new_set.did, new_set.id_data, utoken, tenant_token)
+
+        dev.authsets.append(new_set)
+
+    dev.status = "pending"
+
+    return dev
+
+
+def make_accepted_device_with_multiple_authsets(
+    utoken, num_auth_sets=1, num_accepted=1, tenant_token=""
+):
```

Comment:
Just curious, but what is the motivation for renaming this helper? The other ones (`make_pending_device`, for instance) also seem to work with multiple authsets.

Reply:
+1

Reply:
there is already `make_accepted_device` method in the testutils.common and the tests imports it so I had a method name conflict

Reply:
and with that method, it's not possible to create a device with multiple authsets